### PR TITLE
Code fix on tests for `cupyx.scipy.ndiamge` stats functions

### PR DESCRIPTION
**Pull request by takagi**
Follows #3259. Merge #3425 first.

This PR makes some cosmetic changes on tests for `cupyx.scipy.ndiamge` stats functions.

### CODE REVIEW

File: cupyx/scipy/ndimage/measurements.py [hunk 1]
```diff
@@ -305,7 +305,7 @@ def variance(input, labels=None, index=None):
     if not isinstance(input, cupy.ndarray):
         raise TypeError('input must be cupy.ndarray')
 
-    if input.dtype in (cupy.bool_, cupy.complex64, cupy.complex128):
+    if input.dtype in (cupy.complex64, cupy.complex128):
```
Comment (leofang) [inline cupyx/scipy/ndimage/measurements.py:308]:
```suggestion
    if input.dtype.kind == 'c':
```

File: cupyx/scipy/ndimage/measurements.py [hunk 2]
```diff
@@ -380,7 +380,7 @@ def sum(input, labels=None, index=None):
...
```
Comment (leofang) [inline cupyx/scipy/ndimage/measurements.py:383]:
```suggestion
    if input.dtype.kind == 'c':
```

File: cupyx/scipy/ndimage/measurements.py [hunk 3]
```diff
@@ -444,7 +445,7 @@ def mean(input, labels=None, index=None):
...
```
Comment (leofang) [inline cupyx/scipy/ndimage/measurements.py:448]:
```suggestion
    if input.dtype.kind == 'c':
```

File: tests/cupyx_tests/scipy_tests/ndimage_tests/test_measurements.py [hunk 4]
```diff
@@ -105,83 +105,119 @@ def test_label_swirl(self, xp, scp):
...
+        return op(image, labels, index)
+
+    @testing.for_all_dtypes(no_complex=True)
+    @testing.numpy_cupy_array_equal(scipy_name='scp')
+    def test_ndimage_zero_dim(self, xp, scp, dtype):
+        image = self._make_image((), xp, dtype)
+        label = testing.shaped_random((), xp, dtype=xp.int32, scale=4)
+        index = xp.array([1, 2, 3])
+        op = getattr(scp.ndimage, self.op)
+        return op(image, label, index)
```
Comment (leofang) [inline tests/cupyx_tests/scipy_tests/ndimage_tests/test_measurements.py:150]: It seems this function is a duplicate of the above one?

File: tests/cupyx_tests/scipy_tests/ndimage_tests/test_measurements.py [hunk 5]
```diff
@@ -105,83 +105,119 @@ def test_label_swirl(self, xp, scp):
...
-    def test_ndimage_wrong_dtype(self, dtype):
-        image = cupy.arange(100).astype(dtype)
-        label = cupy.random.randint(1, 4, dtype=cupy.int32)
+    def test_scalar_index(self, xp, scp, dtype):
+        image = self._make_image((100,), xp, dtype)
+        labels = testing.shaped_random((100,), xp, dtype=xp.int32, scale=4)
+        op = getattr(scp.ndimage, self.op)
+        return op(image, labels, 1)
+
+    @testing.for_dtypes([cupy.complex64, cupy.complex128])
```
Comment (leofang) [inline tests/cupyx_tests/scipy_tests/ndimage_tests/test_measurements.py:175]: nitpick:
```suggestion
    @testing.for_complex_dtypes()
```

File: tests/cupyx_tests/scipy_tests/ndimage_tests/test_measurements.py (hunk 4 above)
Comment (takagi) [reply tests/cupyx_tests/scipy_tests/ndimage_tests/test_measurements.py:150]: Let me remember what I wanted to do...

Comment (takagi) [reply tests/cupyx_tests/scipy_tests/ndimage_tests/test_measurements.py]: I removed the duplicate test.

**Comment by kmaehashi** [comment]
pfnCI, test this please.

**Comment by takagi** [comment]
I've merged the master on this, would you kick the CI again?

**Comment by kmaehashi** [comment]
pfnCI, test this please.

**Pull request merged**